Frends.FTP.UploadFiles/Frends.FTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -0,0 +1,379 @@
+﻿using FluentFTP;
+using Frends.FTP.UploadFiles.Logging;
+using Frends.FTP.UploadFiles.TaskConfiguration;
+using Frends.FTP.UploadFiles.TaskResult;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using Frends.FTP.UploadFiles.Enums;
+
+namespace Frends.FTP.UploadFiles.Definitions
+{
+    /// <summary>
+    /// Main class for FTP file transfers
+    /// </summary>
+    internal class FileTransporter
+    {
+        private readonly IFtpLogger _logger;
+        private readonly BatchContext _batchContext;
+        private readonly string[] _filePaths;
+        private readonly RenamingPolicy _renamingPolicy;
+
+        /// <summary>
+        ///     Constructor for SFTP file transfers
+        /// </summary>
+        public FileTransporter(IFtpLogger logger, BatchContext context, Guid instanceId)
+        {
+            _logger = logger;
+            _batchContext = context;
+            _renamingPolicy = new RenamingPolicy(_batchContext.Info.TransferName, instanceId);
+
+            Result = new List<SingleFileTransferResult>();
+            _filePaths = ConvertObjectToStringArray(context.Source.FilePaths);
+        }
+
+        /// <summary>
+        /// List of transfer results.
+        /// </summary>
+        private List<SingleFileTransferResult> Result { get; }
+
+        /// <summary>
+        /// Executes file transfers
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        /// <exception cref="DirectoryNotFoundException"></exception>
+        /// <exception cref="FileNotFoundException"></exception>
+        /// <exception cref="ArgumentException"></exception>
+        public FileTransferResult Run(CancellationToken cancellationToken)
+        {
+            var userResultMessage = "";
+            try
+            {
+                // Fetch source file info and check if files were returned.
+                var (files, success) = GetSourceFiles(_batchContext.Source);
```

Comment:
Source directory doesn't handle macros. IMO Cobalt handles macros with filename and both source and destination directories.

Reply:
According to docs source filenames are not handled I think? Otherwise yes, directories fixed